### PR TITLE
feat: Torch can be turnt on.

### PR DESCRIPTION
**Pull request by csutorasr**
The #58 could be resolved with this.

### CODE REVIEW

File: src/app/modules/zxing-scanner/browser-code-reader.ts [hunk 1]
```diff
@@ -137,6 +156,27 @@ export class BrowserCodeReader {
         this.videoElement.addEventListener('loadedmetadata', this.videoLoadedMetadataEventListener);
     }
 
+    private checkTorchCompatibility(): void {
```
Comment (odahcam) [inline src/app/modules/zxing-scanner/browser-code-reader.ts]: Should return something, a Observable, Promise, something to get the compatibility test value.

File: src/app/modules/zxing-scanner/zxing-scanner.component.ts [hunk 2]
```diff
@@ -89,6 +89,14 @@ export class ZXingScannerComponent implements AfterViewInit, OnDestroy, OnChange
     @Input()
     autofocusEnabled = true;
 
+    /**
+     * Allow start scan or not.
+     */
+    @Input()
+    set torch(on: boolean) {
+        this.codeReader.setTorch(on);
```
Comment (odahcam) [inline src/app/modules/zxing-scanner/zxing-scanner.component.ts]: I understad `codeReader` as a input class and turning torch on as a output, that seems strange to me, I'm thinking in put that logic in another class that both component and reader makes use. 🤔

File: src/app/modules/zxing-scanner/browser-code-reader.ts [hunk 3]
```diff
@@ -137,6 +156,27 @@ export class BrowserCodeReader {
...
+    public setTorch(on: boolean): void {
+        if (this.torchCompatible) {
+            this.track.applyConstraints({
+                advanced: [<any>{ torch: on }]
+            });
+        }
+    }
+
+    public get torchAvailable() {
+        return this.torchCompatible;
```
Comment (odahcam) [inline src/app/modules/zxing-scanner/browser-code-reader.ts]: Here we have a syncronous thing depending on a assyncronously setted variable value. Maybe making it all async should make things safier.

File: src/app/modules/zxing-scanner/browser-code-reader.ts [hunk 4]
```diff
@@ -7,6 +7,11 @@ import {
     HTMLCanvasElementLuminanceSource,
 } from '@zxing/library';
 
+declare class ImageCapture {
```
Comment (odahcam) [inline src/app/modules/zxing-scanner/browser-code-reader.ts]: We should put this little guy on its own file (1:1 rule).

File: src/app/modules/zxing-scanner/zxing-scanner.component.ts (hunk 2 above)
Comment (csutorasr) [reply src/app/modules/zxing-scanner/zxing-scanner.component.ts]: I cannot understand what you want. Do you mean a `[(torch)]` (banana in box) binding for the torch value?

Comment (odahcam) [reply src/app/modules/zxing-scanner/zxing-scanner.component.ts]: I'm talking about the `this.codeReader` object.

**Comment by csutorasr** [comment]
Does this need to be in v2.0? This is only a feature without breaking changes, so it can be in v1.1.

**Comment by odahcam** [comment]
It depends on the available time we all have. 😞 

Are you in need of this feature with some urgency?

**Comment by csutorasr** [comment]
Yes, I would use it as soon as possible.

**Comment by werthdavid** [comment]
PR looks fine to me.
I'm ok with merging it as @csutorasr needs it asap. It shouldn't interfere or break anything.
We can refactor codeReader in 2.0 then..

File: src/app/modules/zxing-scanner/zxing-scanner.component.ts (hunk 2 above)
Comment (csutorasr) [reply src/app/modules/zxing-scanner/zxing-scanner.component.ts]: Now I see your point, I will do that in a later PR. (maybe next week)

Comment (odahcam) [reply src/app/modules/zxing-scanner/zxing-scanner.component.ts]: I'm very curious about this `set <property>` approach. Do you think we can refactor things like `setCodeReaderThrottling` to that? 🤔

**Comment by csutorasr** [comment]
Could you please merge this to the master and make it appear as v1.1.0 on npm?

**Comment by odahcam** [comment]
Not today, can you download and test the latest develop branch? The demo doesn't works. :/

**Pull request merged**